### PR TITLE
Add confirmed_position and current_position at cdc job item info.

### PR DESCRIPTION
**Pull request by azexcy**
Related: #22500

Changes proposed in this pull request:
  - Add confirmed_position and current_position at cdc job item info
  - Change LSN display to String, because PG_CURRENT_XLOG_LOCATION return `String`

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by azexcy** [comment]
example:

```
sharding_db=> show streaming status j0302p0000a6b75349aa39dc7957ff5d11e70cbce3;
 item | data_source |          status          | active | processed_records_count | inventory_finished_percentage | incremental_idle_seconds | confirmed_position | current_position | error_message
------+-------------+--------------------------+--------+-------------------------+-------------------------------+--------------------------+--------------------+------------------+---------------
 0    | ds_0        | EXECUTE_INCREMENTAL_TASK | true   | 100000                  | 100                           | 1779                     | 5/588377F0         | 5/58839E48       |
 1    | ds_1        | EXECUTE_INCREMENTAL_TASK | true   | 10000                   | 100                           | 1779                     | 2/3A9AB968         | 2/3AE1D408       |
(2 rows)
```


query current position:

```
opengauss=# select * from PG_CURRENT_XLOG_LOCATION();
 pg_current_xlog_location
--------------------------
 5/58839EC8
(1 row)

opengauss=#
```

File: kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/OpenGaussPositionInitializer.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public WALPosition init(final DataSource dataSource, final String slotNameSuffix
     
     @Override
     public WALPosition init(final String data) {
-        return new WALPosition(new OpenGaussLogSequenceNumber(LogSequenceNumber.valueOf(Long.parseLong(data))));
+        return new WALPosition(new OpenGaussLogSequenceNumber(LogSequenceNumber.valueOf(data)));
```
Comment (sandynz) [inline kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/OpenGaussPositionInitializer.java:59]: Could `asLong()` in `BaseLogSequenceNumber` and sub-classes be removed?

File: kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/decode/OpenGaussLogSequenceNumber.java [hunk 2]
```diff
@@ -40,4 +38,9 @@ public long asLong() {
     public Object get() {
         return logSequenceNumber;
     }
+    
+    @Override
+    public String toString() {
+        return logSequenceNumber.asString();
+    }
```
Comment (sandynz) [inline kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/decode/OpenGaussLogSequenceNumber.java:45]: Could we use `asString` method name?

File: kernel/data-pipeline/scenario/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/pojo/CDCJobItemInfo.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    private final int inventoryFinishedPercentage;
+    
+    private final JobItemIncrementalTasksProgress incremental;
+    
+    private final String confirmedPosition;
+    
+    private final String currentPosition;
+    
+    private final String errorMessage;
+}
```
Comment (sandynz) [inline kernel/data-pipeline/scenario/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/pojo/CDCJobItemInfo.java:55]: Could we embed `TransmissionJobItemInfo` and just keep 2 new fields (`confirmedPosition` and `currentPosition`)?

**Review by sandynz — CHANGES_REQUESTED** [review]

**Review by sandynz — APPROVED** [review]

**Pull request merged**